framework/db/ActiveRelationTrait.php

```diff
@@ -523,7 +523,7 @@ private function filterByModels($models)
             // single key
             $attribute = reset($this->link);
             foreach ($models as $model) {
-                if (($value = $model[$attribute]) !== null) {
```

Comment:
Looking at the code I don't see anything that would fail on PHP 7 but won't fail on PHP 5. Are you sure it's PHP 7 specific?

Reply:
@samdark, yes, I can confirm it's php 7.4 incompatible change: https://www.php.net/manual/en/migration74.incompatible.php
>>> Array-style access of non-arrays
>>> Trying to use values of type null, bool, int, float or resource as an array (such as $null["key"]) will now generate a notice.



+ #17914

Please, merge this PR.

Reply:
Do I need create another PR or you can merge this one?

Reply:
@Ximich How can you get `null` in `$models`?

Reply:
@rob006, in relation with via, where relation attribute has null value.

```
    public function getSomething()
    {
        return $this->hasOne(Something::class, ['id' => 'something_id'])->via('viaAnotherModel');
    }
```

This code works fine in php 7.3.

Reply:
populateRelation() method in ActiveRelationTrait return $viaModels with null value in this case here: https://github.com/yiisoft/yii2/blob/master/framework/db/ActiveRelationTrait.php#L245

Reply:
Shouldn't we remove these nulls from `$models` before calling `filterByModels()`? IMO it looks weird that this method needs to handle nulls, while it obviously expecting models.

Reply:
I don't know where's right place to filter null models. @samdark, what do you think: should we filter null `$models` in `populateRelation()` method or check it in `filterByModels()`?

Reply:
Filtering sounds like a right thing to do.